src/zarr/core/array.py

```diff
@@ -3785,7 +3797,11 @@ def _get_default_codecs(
     | Literal["auto"]
     | None
 )
-CompressorLike: TypeAlias = dict[str, JSON] | BytesBytesCodec | numcodecs.abc.Codec | None
+# Union of acceptable types for v2 and v3 compressors
+CompressorLike: TypeAlias = (
```

Comment:
Do we want `None` in here? I feel like `CompressorLike` should be all the different ways of specifying a particular compressor, but `None` is not that.

Reply:
`None` is an acceptable compressor for v2

Reply:
`None` is an acceptable value for the `"compressor"` field in metadata, but it's not a compressor. that's the distinction i'm getting at

Reply:
and if we were to write a type annotation for the `"compressor"` field in metadata, it would be `Compressor | None`

Reply:
Ah, I see what you mean. This is intended as a type annotation for the public API, which means `None` and 'auto' are acceptable values. I updated the code comment to clarify this a bit.

Reply:
Yes and my suggestion is about the public API. We should use `CompressorLike` for something that is like a compressor. That's compatible with a function that takes `CompressorLike | Literal["auto"] | None`.

Reply:
I had a quick look at what other existing `*Like` types do, and the ones already in use with `Group.create_array` (e.g., `ShardsLike`, `FiltersLike`) also contain `'auto'` and `None`. So while I don't disagree with changing that, I think that's something that should be done consistently and is outside the scope of this PR?

Reply:
I don't agree that we need specific typevar for this type, as long as we can use a short union, but this is probably a discussion for another PR entirely

Reply:
> I had a quick look at what other existing `*Like` types do, and the ones already in use with `Group.create_array` (e.g., `ShardsLike`, `FiltersLike`) also contain `'auto'` and `None`. So while I don't disagree with changing that, I think that's something that should be done consistently wiht all the types and is outside the scope of this PR?

agreed! sorry for the noise